tests/util/pool_tests.rs

```diff
@@ -61,6 +68,78 @@ pub fn test_thinpool_expand(paths: &[&Path]) -> () {
 }
 
 
+/// Verify a snapshot has the same files and same contents as the origin.
+pub fn test_filesystem_snapshot(paths: &[&Path]) {
+    let dm = DM::new().unwrap();
+    let (mut pool, _) =
+        StratPool::initialize("stratis_test_pool", &dm, paths, Redundancy::NONE, true).unwrap();
+    let &(_, fs_uuid) = pool.create_filesystems(&[("stratis_test_filesystem", None)])
+        .unwrap()
+        .first()
+        .unwrap();
+    let write_buf = &[8u8; SECTOR_SIZE];
+    let file_count = 10;
+    let source_tmp_dir = TempDir::new("stratis_testing").unwrap();
+    {
+        // to allow mutable borrow of pool
+        let filesystem = pool.get_mut_strat_filesystem(&fs_uuid).unwrap();
+        println!("Mount {:?} on {:?}",
+                 filesystem.devnode(),
+                 source_tmp_dir.path());
+        mount(Some(&filesystem.devnode()),
+              source_tmp_dir.path(),
+              Some("xfs"),
+              MS_VERBOSE,
+              None as Option<&str>)
+                .unwrap();
+        for i in 0..file_count {
+            let file_path = source_tmp_dir
+                .path()
+                .join(format!("stratis_test{}.txt", i));
+            let mut f = OpenOptions::new()
+                .create(true)
+                .write(true)
+                .open(file_path)
+                .unwrap();
+            f.write_all(write_buf).expect("write to file");
+            f.flush().expect("flush snapshot source");
+
+        }
+
+    }
+    // Run a check to expand the pool.  This is necessary because
```

Comment:
I absolutely don't understand why running ```pool.check()``` has some important effect.

Reply:
@mulkieran would this comment make it clear?

    // Run a check to expand the pool. The space initially allocated
    // to a pool is close to consumed by the filesystem and few files
    // written above. If we attempt to update the UUID on the snapshot
    // without expanding the pool, the pool will go into out-of-data-space
    // (queue IO) mode, causing the test to fail. Calling pool.check() will
    // compare the data space used by the pool to the data space allocated
    // and expand when there is less than DATA_LOWATER space remaining..

Reply:
Yes. I understand what's going on now. Could you also add the following:

```// TODO: Make use of a way (not as yet existing) to explicitly extend pool to the necessary size.```

?

Reply:
ok - done.